ICRCs/ICRC-107/ICRC-107.md

```diff
@@ -0,0 +1,206 @@
+
+ICRC-107 introduces a new block type that follows the ICRC-3 specification to configure `icrc107_fee_col`.
+
+In addition to the ICRC-3 specific requirements, a fee collector configuration block:
+
+- **MUST** contain a field `btype` with value `"107feecol"`.
+- **MUST** contain a field `icrc107_fee_col` of type `Array`, specifying the owner and, optionally, a subaccount, following the ICRC-3 standard.
+- To explicitly burn fees, `icrc107_fee_col` **MUST** be set to `variant { Array = vec {}}`.
+
+#### Example Block
+
+Below is an example of a valid **fee collector configuration block** that sets the fee collector to a specific account:
+
+```
+variant { Map = vec {
+    // Block type: indicates this is a fee collector configuration block
+    record { "btype"; variant { Text = "107feecol" }};
+    // Fee collector account: specifies the account that will collect fees
+    record { "icrc107_fee_col"; variant { Array = vec {
+        variant { Blob = blob "\00\00\00\00\02\00\01\0d\01\01"}; // Owner principal
+        variant { Blob = blob "\06\ec\cd\3a\97\fb\a8\5f\bc\8d\a3\3e\5d\ba\bc\2f\38\69\60\5d\c7\a1\c9\53\1f\70\a3\66\c5\a7\e4\21" }; // Subaccount
+    }} };
+    // Timestamp: indicates when the block was created
+    record { "ts"; variant { Nat = 1_741_312_737_184_874_392 : nat } };
+    // Parent hash: links this block to the previous block in the chain
+    record { "phash";
+              variant {
+                Blob = blob "\d5\c7\eb\57\a2\4e\fa\d4\8b\d1\cc\54\9e\49\c6\9f\d1\93\8d\e8\02\d4\60\65\e2\f2\3c\00\04\3b\2e\51"
+              }};
+}};
+```
+
+A block that explicitly sets fee burning by removing the fee collector (i.e., all fees are burned from this point onward):
+
+```
+variant { Map = vec {
+    record { "btype"; variant { Text = "107feecol" }};
+    record { "icrc107_fee_col"; variant { Array = vec {
+            }}};
+    record { "ts"; variant { Nat = 1_741_312_737_184_874_392 : nat } };
+    record { "phash";
+              variant {
+                Blob = blob "\2d\86\7f\34\c7\2d\1e\2d\00\84\10\a4\00\b0\b6\4c\3e\02\96\c9\e8\55\6f\dd\72\68\e8\df\8d\8e\8a\ee"
+              }};
+}};
+```
+
+
+
+---
+
+# ICRC-107: Methods for Setting and Getting Fee Collection
+
+## 3. Methods for Setting and Getting Fee Collection
+
+### 3.1 `icrc107_set_fee_collection`
+
+This method allows a ledger controller to update the fee collection settings. It modifies the `icrc107_fee_col` account, which determines where collected fees are sent. The updated settings are recorded in the next block added to the ledger.
+
+```
+icrc107_set_fee_collection: (SetFeeCollectionRequest) -> ();
+```
+
+#### Request Type:
+```
+type Account = record {
+    owner: principal;
+    subaccount: opt blob;
+};
+
+type SetFeeCollectionRequest = record {
+    icrc107_fee_col: opt Account;
+};
+```
+
+This method MUST only be callable by the ledger controller or authorized principals. Unauthorized calls SHOULD result in an error.
```

Comment:
> Unauthorized calls SHOULD result in an error.

SHOULD or MUST? 🤔 

do we need to define the expected error, too?

Reply:
I've now defined the errors explicitly